sw/device/lib/dif/dif_edn.c

```diff
@@ -267,45 +275,51 @@ dif_result_t dif_edn_get_main_state_machine(const dif_edn_t *edn,
 
 dif_result_t dif_edn_instantiate(
     const dif_edn_t *edn, dif_edn_entropy_src_toggle_t entropy_src_enable,
-    const dif_edn_seed_material_t *seed_material) {
-  if (edn == NULL) {
+    const dif_edn_cmd_t *cmd) {
+  if (edn == NULL || cmd == NULL) {
     return kDifBadArg;
   }
+  dif_csrng_seed_material_t csrng_seed_material;
+  csrng_seed_material.seed_material_len = cmd->len;
+  memcpy(&csrng_seed_material.seed_material, cmd->data,
```

Comment:
This doesn't seem right. `csrng_seed_material.seed_material` is a pointer already. So I would expect either `csrng_seed_material.seed_material` or `&csrng_seed_material.seed_material[0]`.

Reply:
I am not sure you are correct or not, because seed_material is a struct holding a length + the data array. Anyway, as per @moidx 's suggestion, I've now reworked the PR to reduce the number of changes. This allowed dropping the changes of question here. I'll resolve the remaining discussion points.